pdns/dnsdist-lua.hh

```diff
@@ -107,3 +107,37 @@ void setupLuaInspection(LuaContext& luaCtx);
 void setupLuaVars(LuaContext& luaCtx);
 void setupLuaWeb(LuaContext& luaCtx);
 void setupLuaLoadBalancingContext(LuaContext& luaCtx);
+
+/**
+ * getOptionalValue(vars, key, value)
+ *
+ * Attempts to extract value for key in vars.
+ * Erases the key from vars.
+ *
+ * returns: -1 if type wasn't compatible, 0 if not found or number of element(s) found
+ */
+template<class G, class T, class V>
+static inline int getOptionalValue(boost::optional<V>& vars, const std::string& key, T& value) {
+  /* nothing found, nothing to return */
+  if (!vars)
+    return 0;
+  if (vars->count(key)) {
+    try {
+      value = boost::get<G>((*vars)[key]);
+    } catch (boost::bad_get& e) {
+      /* key is there but isn't compatible */
+      return -1;
+    }
+  }
+  return vars->erase(key);
+}
+
+template<class V>
+static inline void checkAllParametersConsumed(const boost::optional<V>& vars) {
```

Comment:
Did you consider passing a `const std::string&` indicating where the parameters were not consumed? It would be a bit of extra work but I think it might be very helpful, so I'm willing to write that code if that makes sense to you :)

Reply:
What would the string do?

Reply:
Tell what directive was involved, like `newServer` or `addLocal`?